model/config.go

```diff
@@ -473,3 +474,44 @@ func (store *istioConfigStore) Policy(instances []*ServiceInstance, destination
 
 	return out
 }
+
+// this function rejects rules that have a domain which same to domains of some other rule.
+// According to Envoy's virtual host specification, no virtual hosts can share the same domain.
+// The following code rejects conflicting rules determenistically, by a lexicographical order - a rule with
+// a smaller name lexicographically wins.
+func RejectConflictingEgressRules(egressRules map[string]*proxyconfig.EgressRule) (map[string]*proxyconfig.EgressRule, error) {
+	filteredEgressRules := make(map[string]*proxyconfig.EgressRule)
+	var errs error
+
+	var keys []string
+	for key, _ := range egressRules {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	domains := make(map[string]string)
+	for _, key := range keys {
+		egressRule := egressRules[key]
+		conflictingRule := false
+		for _, domain := range egressRule.Domains {
+			keyOfAnEgressRuleWithTheSameDomain, conflictFound := domains[domain]
```

Comment:
reuse `conflictingRule` instead of defining a new bool (and setting `conflictingRule` to true seperately) ?